### ISSUE TITLE
Slow mocks shutdown in Go 1.15

### ISSUE THREAD

**Issue by What-If-I** [lamoda/gonkey#57]
In Go 1.15 behavior of http.Server.Shutdown [changed](https://github.com/golang/go/commit/5a75f7c0b0789fe04ea4879a524cc95dbe734636#diff-bf538ad0b0c9263061db13ca6d8f324eR2683) and now server waits for all listeners to disconnect.
```go
// go/1.15/libexec/src/net/http/server.go:2699
...
ticker := time.NewTicker(shutdownPollInterval) // 500 ms
defer ticker.Stop()
for {
	if srv.closeIdleConns() && srv.numListeners() == 0 { // for some reason this is always false first time
		return lnerr
	}
	select {
	case <-ctx.Done():
		return ctx.Err()
	case <-ticker.C:  // we always wait here half a second
	}
...
``` 

For some reason, on mock shutdown we always still have some active connections and shutdown takes about 500ms. And since we [shutdown all mocks synchronously](https://github.com/lamoda/gonkey/blob/master/mocks/mocks.go#L44-L48) with a lot of mocks we wait for tests to finish for up to **18 seconds**.

Does anyone else have this issue?

As a solution we can close each mock server in a separate goroutine, that should reduce wait time for us from 18s to 500ms.

**Comment by luza** [comment]
Suggested solution looks more like a work around. I've better to find out why it can be shut down immediately as I don't see any reason to wait even 500 ms.

**Comment by c1rno** [comment]
Look at this PR https://github.com/lamoda/gonkey/pull/58

**Comment by What-If-I** [comment]
Fixed in #59
